Reporting/ad-extensions-reporter.js

```diff
@@ -0,0 +1,395 @@
+    );
+    var reportRows = placeholderFeedItemReport.rows();
+
+    // Process report data, to prepare for outputting to the spreadsheet
+    Logger.log("Processing report data");
+    var extensionPlaceholderTypeIds = Object.keys(extensionPlaceholderTypeIdToFieldIdsObjectMapping);
+
+    var extensionTypeFeedItemCounts = {};
+    for (var i = 0, n = extensionPlaceholderTypeIds.length; i < n; i++) {
+        extensionTypeFeedItemCounts[extensionPlaceholderTypeIds[i]] = 0;
+    }
+    var jsonAttributeValuesByExtensionType = {};
+    var disapprovalShortNamesByExtensionType = {};
+    var duplicateAttributeValuesCounts = {};
+
+    while (reportRows.hasNext()) {
+        var row = reportRows.next();
+
+        var placeholderTypeId = row['PlaceholderType']
+        if (extensionPlaceholderTypeIds.indexOf(placeholderTypeId) === -1) {
+            continue;
+        }
+        if (Object.keys(jsonAttributeValuesByExtensionType).indexOf(placeholderTypeId) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId] = [];
+            disapprovalShortNamesByExtensionType[placeholderTypeId] = [];
+        }
+
+        var jsonAttributeValues = row['AttributeValues'];
+        var jsonDisapprovalShortNames = row['DisapprovalShortNames'];
+        if ('' === jsonDisapprovalShortNames.toString()) {
+            var disapprovalShortNames = [];
+        } else {
+            var disapprovalShortNames = JSON.parse(jsonDisapprovalShortNames);
+        };
+        if (jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues) === -1) {
+            jsonAttributeValuesByExtensionType[placeholderTypeId].push(jsonAttributeValues);
+            disapprovalShortNamesByExtensionType[placeholderTypeId].push(disapprovalShortNames);
+        } else {
+            if (Object.keys(duplicateAttributeValuesCounts).indexOf(jsonAttributeValues) === -1) {
+                duplicateAttributeValuesCounts[jsonAttributeValues] = 2;
+            } else {
+                duplicateAttributeValuesCounts[jsonAttributeValues] += 1;
+            };
+            for (var i = 0, n = disapprovalShortNames.length; i < n; i++) {
+                var disapprovalName = disapprovalShortNames[i];
+                if (disapprovalShortNamesByExtensionType[placeholderTypeId][
+                    jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                ].indexOf(disapprovalName) === -1) {
+                    disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ] = disapprovalShortNamesByExtensionType[placeholderTypeId][
+                        jsonAttributeValuesByExtensionType[placeholderTypeId].indexOf(jsonAttributeValues)
+                    ].concat(disapprovalName);
+                }
+            }
+        }
+        extensionTypeFeedItemCounts[placeholderTypeId] += 1;
```

Comment:
Could probably eliminate this variable and then work it out at the end, by using `jsonAttributeValuesByExtensionType`?

Reply:
Yes!